### PR TITLE
feat(connectors): add `connectors list-available` command

### PR DESCRIPTION
**Pull request by Paveltarno**
> [!NOTE]
> ## Description
>
> This PR adds a new `connectors list-available` subcommand that fetches and displays all available integration types from the Base44 API. Each integration is rendered in human-readable YAML format with a reusable `formatYaml` helper that strips empty/null values by default. The changes align with the existing connectors command pattern and include full test coverage for both the command and the YAML utility.
>
> ## Related Issue
>
> None
>
> ## Type of Change
>
> - [ ] Bug fix (non-breaking change which fixes an issue)
> - [x] New feature (non-breaking change which adds functionality)
> - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
> - [ ] Documentation update
> - [ ] Refactoring (no functional changes)
> - [ ] Other (please describe):
>
> ## Changes Made
>
> - Added `connectors list-available` command that calls `GET /external-auth/available-integrations` and prints each integration as YAML
> - Added `listAvailableIntegrations()` API function in `core/resources/connector/api.ts` with Zod validation and snake_case→camelCase transform
> - Added `ListAvailableIntegrationsResponseSchema` and `ConnectionConfigFieldSchema` to `connector/schema.ts`
> - Added `formatYaml()` helper in `packages/cli/src/cli/utils/yaml.ts` with configurable empty-value stripping using `lodash.isEmpty`
> - Registered `list-available` as a subcommand on the existing `connectors` command
> - Added `yaml` package dependency and updated `bun.lock`
> - Added integration tests for all edge cases (success, empty list, API error, invalid data, missing project)
> - Added unit tests for `formatYaml` covering null/undefined/empty stripping and nested objects
> - Removed `packages/cli/plan.md` (working notes, not needed in repo)
> - Added `.claude/worktrees/` to `.gitignore`
>
> ## Testing
>
> - [ ] I have tested these changes locally
> - [x] I have added/updated tests as needed
> - [ ] All tests pass (`npm test`)
>
> ## Checklist
>
> - [x] My code follows the project's style guidelines
> - [x] I have performed a self-review of my own code
> - [ ] I have commented my code, particularly in hard-to-understand areas
> - [ ] I have made corresponding changes to the documentation (if applicable)
> - [x] My changes generate no new warnings
> - [ ] I have updated `docs/` (AGENTS.md) if I made architectural changes
>
> ## Additional Notes
>
> The `AvailableIntegrationSchema` intentionally uses `z.string()` instead of `IntegrationTypeSchema` so the command works with integration types not yet defined in the CLI's enum — this ensures forward compatibility as new integrations are added on the server side.
>
> ---
> <sub>🤖 Generated by Claude | 2026-03-09 00:00 UTC</sub>

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -80,6 +80,7 @@
     "tmp-promise": "^3.0.3",
     "typescript": "^5.7.2",
     "vitest": "^4.0.16",
+    "yaml": "^2.8.2",
```
Comment (Paveltarno) [inline package.json]: added this yaml dep, i think it's worth it:
1. we should probably just start formatting yaml as part of our regular clack output, it integrates pretty well and we don't need to manually fiddle with formatting our domain entities
2. we might just add `--output yaml` with no extra cost

File: packages/cli/src/cli/commands/connectors/list-available.ts [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+      successMessage: "Available integrations fetched successfully",
+      errorMessage: "Failed to fetch available integrations",
+    },
+  );
+
+  if (integrations.length === 0) {
+    return { outroMessage: "No available integrations found." };
+  }
+
+  for (const i of integrations) {
```
Comment (artemdemo) [inline packages/cli/src/cli/commands/connectors/list-available.ts]: `i` is also usual name in `for` loops and very general, hence confusing in this case.
Maybe rename it to `integration`?

File: packages/cli/src/cli/commands/connectors/list-available.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    const data: Record<string, unknown> = {
+      type: i.integrationType,
+      description: i.description,
+    };
```
Comment (artemdemo) [inline packages/cli/src/cli/commands/connectors/list-available.ts]: Can't we define specific type for it and not just general one?

File: packages/cli/src/cli/commands/connectors/list-available.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+      data.connectionConfigFields = i.connectionConfigFields.map((f) => {
+        const field: Record<string, unknown> = {};
+        for (const [k, v] of Object.entries(f)) {
+          if (v != null) field[k] = v;
+        }
+        return field;
+      });
+    }
+    const yaml = stringify(data, { indent: 2 }).trimEnd();
+    log.info(`${i.displayName}\n  ${yaml.replace(/\n/g, "\n  ")}`);
```
Comment (artemdemo) [inline packages/cli/src/cli/commands/connectors/list-available.ts]: Then maybe this approach should be wrapped in some helper function like `yamlFormatting`, so `replace` can be part of it and we wouldn't need to copy-paste it when will use it again.
so usage will be something like:
```
log.info(`${i.displayName}\n  ${yamlFormatting(data)}`);
```

File: packages/cli/src/cli/commands/connectors/list-available.ts (hunk 2 above)
Comment (Paveltarno) [reply packages/cli/src/cli/commands/connectors/list-available.ts]: @claude do your thing

Comment (kfirstri) [reply packages/cli/src/cli/commands/connectors/list-available.ts]: I don't think it did anything

Comment (Paveltarno) [reply packages/cli/src/cli/commands/connectors/list-available.ts]: קלוד לא חבר

Comment (Paveltarno) [reply packages/cli/src/cli/commands/connectors/list-available.ts]: Fixed — renamed `i` to `integration`.

File: packages/cli/src/cli/commands/connectors/list-available.ts (hunk 3 above)
Comment (Paveltarno) [reply packages/cli/src/cli/commands/connectors/list-available.ts]: removed the type alltogether

File: packages/cli/src/cli/commands/connectors/list-available.ts [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+  );
+
+  if (integrations.length === 0) {
+    return { outroMessage: "No available integrations found." };
+  }
+
+  for (const { displayName, ...rest } of integrations) {
+    const yaml = formatYaml(rest);
+    const pad = " ".repeat(YAML_INDENT);
+    log.info(`${displayName}\n${pad}${yaml.replace(/\n/g, `\n${pad}`)}`);
```
Comment (artemdemo) [inline packages/cli/src/cli/commands/connectors/list-available.ts:32]: @Paveltarno why this part is not happening in `formatYaml`?

Comment (artemdemo) [reply packages/cli/src/cli/commands/connectors/list-available.ts:32]: I'm referring to padding.

Comment (Paveltarno) [reply packages/cli/src/cli/commands/connectors/list-available.ts:32]: good question, this is because the yamlformatter only formats objects into yaml
but the whole output is indented because we also have the `displayName` as a header

Screenshot 2026-03-09 at 18 49 55

this is nothing to do with the yaml formatter, more with how the CLI decides to embed that format into it's output

Comment (artemdemo) [reply packages/cli/src/cli/commands/connectors/list-available.ts:32]: I see, thank you for clarification
